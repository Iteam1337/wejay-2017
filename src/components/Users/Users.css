.Users {
  display: grid;
  grid-gap: 5px;
  grid-template-columns: repeat(auto-fill, minmax(30px, 1fr));
  margin-top: 30px; }
