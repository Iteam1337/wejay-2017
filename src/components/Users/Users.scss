.Users {
  margin-top: 30px;
}

.Users__user {
  align-items: center;
  background-color: rgba(102, 140, 255, 0.08);
  border-radius: 4px;
  display: flex;
  padding: 10px 20px;

  &:not(:last-child) {
    margin-bottom: 10px;
  }
}

.Users__email {
  color: #666;
  flex: 1;
  margin-left: 20px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.Users__lastPlay {
  font-size: 12px;
}
