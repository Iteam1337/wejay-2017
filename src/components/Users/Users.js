import './Users.css'
import React from 'react'
import Gravatar from 'components/Gravatar'

const Users = ({ users }) => {
  if (!users.length) {
    return null
  }

  return (
    <ul className="Users">
      {users.map(user => (
        <li className="Users__user" key={user.id}>
          <Gravatar email={user.email} size={30} />
        </li>
      ))}
    </ul>
  )
}

export default Users
