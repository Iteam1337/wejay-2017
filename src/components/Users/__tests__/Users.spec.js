import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Users from '../Users'

describe('components/Users', () => {
  let component

  beforeEach(() => {
    component = shallow(
      <Users users={[{ email: 'cookie@monster.com', id: '1' }]} />
    )
  })

  it('returns null if no users', () => {
    component.setProps({
      users: []
    })

    expect(component.type()).toBeNull()
  })

  it('renders Users', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
