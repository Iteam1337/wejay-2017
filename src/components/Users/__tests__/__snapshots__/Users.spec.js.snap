// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Users renders Users 1`] = `
<ul
  className="Users"
>
  <li
    className="Users__user"
  >
    <Gravatar
      email="cookie@monster.com"
      size={30}
    />
    <div
      className="Users__email"
    >
      cookie@monster.com
    </div>
  </li>
</ul>
`;
